extensions/objects/templates/FilepathEditor.js

```diff
@@ -73,27 +162,321 @@ oppia.directive('filepathEditor', [
+            $scope.croppingArea.x1 = Math.min(
+              Math.max(0, x),
+              $scope.croppingArea.x2 - CROP_AREA_MIN_WIDTH_PX);
+          };
+
+          var adjustResizeRight = function(x) {
+            // Update crop area x2 value, correcting for boundaries.
+            $scope.croppingArea.x2 = Math.max(
+              Math.min(dimensions.width, x),
+              CROP_AREA_MIN_WIDTH_PX + $scope.croppingArea.x1);
+          };
+
+          var adjustResizeTop = function(y) {
+            // Update crop area y1 value, correcting for boundaries.
+            $scope.croppingArea.y1 = Math.min(
+              Math.max(0, y),
+              $scope.croppingArea.y2 - CROP_AREA_MIN_HEIGHT_PX);
+          };
+
+          var adjustResizeBottom = function(y) {
+            // Update crop area y2 value, correcting for boundaries.
+            $scope.croppingArea.y2 = Math.max(
+              Math.min(dimensions.height, y),
+              CROP_AREA_MIN_HEIGHT_PX + $scope.croppingArea.y1);
+          };
+
+          switch (direction) {
+            case MOUSE_TOP_LEFT:
+              adjustResizeTop(y);
+              adjustResizeLeft(x);
+              break;
+            case MOUSE_TOP:
+              adjustResizeTop(y);
+              break;
+            case MOUSE_TOP_RIGHT:
+              adjustResizeTop(y);
+              adjustResizeRight(x);
+              break;
+            case MOUSE_RIGHT:
+              adjustResizeRight(x);
+              break;
+            case MOUSE_BOTTOM_RIGHT:
+              adjustResizeBottom(y);
+              adjustResizeRight(x);
+              break;
+            case MOUSE_BOTTOM:
+              adjustResizeBottom(y);
+              break;
+            case MOUSE_BOTTOM_LEFT:
+              adjustResizeBottom(y);
+              adjustResizeLeft(x);
+              break;
+            case MOUSE_LEFT:
+              adjustResizeLeft(x);
+              break;
+          }
+        };
+
+        $scope.updatePositionWithinCroppingArea = function(x, y) {
+          var margin = CROP_BORDER_MARGIN_PX;
+          var cx1 = $scope.croppingArea.x1;
+          var cy1 = $scope.croppingArea.y1;
+          var cx2 = $scope.croppingArea.x2;
+          var cy2 = $scope.croppingArea.y2;
+
+          var xOnLeftBorder = x > cx1 - margin && x < cx1 + margin;
+          var xOnRightBorder = x > cx2 - margin && x < cx2 + margin;
+          var yOnTopBorder = y > cy1 - margin && y < cy1 + margin;
+          var yOnBottomBorder = y > cy2 - margin && y < cy2 + margin;
+          var xInside = x > cx1 && x < cx2;
+          var yInside = y > cy1 && y < cy2;
+          if (xOnLeftBorder && yOnTopBorder) {
+            // Upper left corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_TOP_LEFT;
+          } else if (xOnLeftBorder && yOnBottomBorder) {
+            // Lower left corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_BOTTOM_LEFT;
+          } else if (xOnLeftBorder) {
```

Comment:
No, you're right. And I looked at the wrong code section. I do clock-wise somewhere else, but here it's kind of random and you SHOULD check the corners first, otherwise we're making the corners smaller than they should be and harder to grab.

Done!

I've added an implementation comment as well.